tinygrad/runtime/ops_cuda.py

```diff
@@ -47,8 +47,21 @@ def __call__(self, global_size, local_size, *args, wait=False):
 class CUDACodegen(CStyleCodegen):
   lang = CStyleLanguage(
     kernel_prefix = "__global__", smem_prefix = "__shared__ ", barrier = "__syncthreads();", float4 = "make_float4",
-    half_prekernel = "#include <cuda_fp16.h>",
     gid = [f'blockDim.{chr(120+i)}*blockIdx.{chr(120+i)}+threadIdx.{chr(120+i)}' for i in range(3)],
-    lid = [f'threadIdx.{chr(120+i)}' for i in range(3)])
+    lid = [f'threadIdx.{chr(120+i)}' for i in range(3)],
+    half_prekernel = """
+      #include <cuda_fp16.h>
+      struct __align__(8) half4 {
+        half2 x, y;
+        __device__ explicit operator float4() const;
+      };
+      __device__ __forceinline__ half4::operator float4() const {return make_float4(__half2float(x.x), __half2float(x.y), __half2float(y.x), __half2float(y.y)); }
+      __device__ __forceinline__ half4 operator+(half4 a, half4 b) { return half4{a.x+b.x, a.y+b.y}; }
```

Comment:
Are these operators used anywhere?

Reply:
they are not, llama tests pass without them. Actually, we can also define the cast operator in the struct directly. my bad.